File: 02 - backend/spring-boot-Inspection-RestfulWebservices/src/main/java/com/mkr/Inspection/requestModels/AddReportRequest.java
package com.mkr.Inspection.requestModels;

import lombok.Data;

import java.util.Date;
@Data
public class AddReportRequest {
    private String projectId;
    private String projectName;
    private String projectType;
    private String inspectorName;
    private String inspectorPhone;
    private String inspectorEmail;
    private Date date;
    private String[] images;
    private String comment;
    private Double progress;
    private Long phase;
    private String excel;
}

File: 03 - frontend/Inspection-React/src/models/AddReportRequest.ts
class InspectionModel {
    projectId: string;
    projectName: string;    
    projectType: string;
    inspectorName: string;
    inspectorPhone: string;
    inspectorEmail?: string;

    date: Date;
    images?: string[];
    excel?: string;
    recipients?: string[];
    comment?: string
    progress?: number
    phase?: number
    constructor(projectId: string, projectName: string, projectType: string, date: Date, recipients: string[],
        inspectorName: string, inspectorPhone: string){
        this.projectId = projectId;
        this.projectName = projectName;
        this.projectType = projectType;
        this.date = date;
        this.recipients = recipients;
        this.inspectorName = inspectorName;
        this.inspectorPhone = inspectorPhone;
    }

    

}
export default InspectionModel;